glean_parser/swift.py

```diff
@@ -121,16 +121,18 @@ def output_swift(objs, output_dir, options={}):
     )
 
     # The object parameters to pass to constructors.
-    # Need to be in the order the type constructor expects them.
+    # **CAUTION**: This list needs to be in the order the type constructor expects them.
+    # The `test_order_of_fields` test checks that the generated code is valid.
+    # **DO NOT CHANGE THE ORDER OR ADD NEW FIELDS IN THE MIDDLE**
     extra_args = [
-        "allowed_extra_keys",
         "category",
-        "disabled",
-        "lifetime",
         "name",
-        "reason_codes",
         "send_in_pings",
+        "lifetime",
+        "disabled",
         "time_unit",
+        "reason_codes",
```

Comment:
Why don't we just take this out here.  We can put it back in right before https://github.com/mozilla/glean/pull/649 is ready to land.

Reply:
Need to test it, that might allow us to land the Swift part of reason codes separately.

Reply:
The real problem with reason codes is currently what code is generated:

```swift
let customPing = Ping<NoExtraKeys>(...)
```

We might just want to skip that `<NoExtraKeys>` part for Swift right now or we need to make glean#649 include the Swift part.

Reply:
Yeah -- I'm working on the Swift part for #649 -- the problem is just what we should land in glean_parser now to make sure glean master is working.

Reply:
Glean master is still on an old parser version, which works. It currently won't work with 1.16.0

Reply:
Oh -- my bad.  I thought we needed to upgrade glean_parser in glean for other reasons.  If not, let's not worry -- we can but a new glean_parser release and update to it as part of #649.

Reply:
👍 Sounds good to me.